### PR TITLE
[cmake] add long routes options

### PR DESCRIPTION
**Pull request by simonlingoogle**
This PR adds `OT_MLE_LONG_ROUTES` option for cmake and `MLE_LONG_ROUTES` option for Makefile.

### CODE REVIEW

**Comment by simonlingoogle** [comment]
More comments on this? @bukepo @jwhui

File: etc/cmake/options.cmake [hunk 1]
```diff
@@ -161,6 +161,11 @@ if(OT_MAC_FILTER)
     list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_MAC_FILTER_ENABLE=1")
 endif()
 
+option(OT_MLE_LONG_ROUTES "enable MLE long routes extension")
```
Comment (jwhui) [inline etc/cmake/options.cmake:164]: Can we add a note here that this breaks Thread conformance, something like:
```suggestion
option(OT_MLE_LONG_ROUTES "enable MLE long routes extension (experimental, breaks Thread conformance)")
```

File: examples/common-switches.mk [hunk 2]
```diff
@@ -199,6 +200,10 @@ ifeq ($(MAC_FILTER),1)
 COMMONCFLAGS                   += -DOPENTHREAD_CONFIG_MAC_FILTER_ENABLE=1
 endif
 
+ifeq ($(MLE_LONG_ROUTES),1)
+COMMONCFLAGS                   += -DOPENTHREAD_CONFIG_MLE_LONG_ROUTES_ENABLE=1
```
Comment (jwhui) [inline examples/common-switches.mk:204]: Add a note here that this breaks Thread conformance?

File: etc/cmake/options.cmake (hunk 1 above)
Comment (simonlingoogle) [reply etc/cmake/options.cmake:164]: Good idea. Will do.

File: examples/common-switches.mk (hunk 2 above)
Comment (simonlingoogle) [reply examples/common-switches.mk:205]: noted

File: etc/cmake/options.cmake (hunk 1 above)
Comment (simonlingoogle) [reply etc/cmake/options.cmake]: noted

**Pull request merged**